host/integration_test.go

```diff
@@ -1017,6 +1010,19 @@ func (s *integrationSuite) TestCronWorkflow() {
 	s.Nil(err)
 	expectedExecutionTime := dweResponse.WorkflowExecutionInfo.GetStartTime() + 3*time.Second.Nanoseconds()
 	s.Equal(expectedExecutionTime, dweResponse.WorkflowExecutionInfo.GetExecutionTime())
+
+	sort.Slice(closedExecutions, func(i, j int) bool {
+		return closedExecutions[i].GetStartTime() < closedExecutions[j].GetStartTime()
+	})
+	// The first execution is the termination execution.
+	lastExecution := closedExecutions[0]
+	for i := 1; i != 4; i++ {
+		executionInfo := closedExecutions[i]
+		backoffDuration := backoff.GetBackoffForNextSchedule(cronSchedule, time.Unix(0, lastExecution.GetStartTime()), time.Unix(0, lastExecution.GetCloseTime()))
+		// Round up to compare only seconds
+		s.Equal(backoffDuration.Nanoseconds()/1000000000, int64(executionInfo.GetExecutionTime()-lastExecution.GetCloseTime())/1000000000)
```

Comment:
`backoffDuration` is basically the `firstDecisionBackoffTime` and when doing describe workflow, the execution time is calculated as `startTime` + `firstDecisionBackoffTime`. So why we use `closeTime` here?

Reply:
When the instance is slow, the close time and start time could be different and we actually use closeTime to calculate the backoff duration

Reply:
Synced offline. The validation should be current-execution execution time - current-execution start time